src/Authorization.AspNetCore/AuthorizationValidationRule.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GraphQL.Language.AST;
+using GraphQL.Types;
+using GraphQL.Validation;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Infrastructure;
+
+namespace GraphQL.Server.Authorization.AspNetCore
+{
+    public class AuthorizationValidationRule : IValidationRule
+    {
+        private readonly IAuthorizationService _authorizationService;
+
+        public AuthorizationValidationRule(IAuthorizationService authorizationService)
+        {
+            _authorizationService = authorizationService;
+        }
+
+        public INodeVisitor Validate(ValidationContext context)
+        {
+            var userContext = context.UserContext as IProvideClaimsPrincipal;
```

Comment:
Since this is always going to be in ASP.NET Core, why not inject `IHttpContextAccessor` and use that to get  `HttpContext.User`? 

Then there's no special requirement for the `UserContext`.

Reply:
Done. Also removed the IProvideClaimsPrincipal interface as I think it's no longer required.